openlibrary/plugins/openlibrary/js/carousels.js

```diff
@@ -59,6 +60,62 @@ const Carousel = {
             slidesToScroll: a,
             responsive: responsive_settings
         });
+
+        var availabilityStatuses = {
+            'open': {cls: 'cta-btn--available', cta: 'Read'},
+            'borrow_available': {cls: 'cta-btn--available', cta: 'Borrow'},
+            'borrow_unavailable': {cls: 'cta-btn--unavailable', cta: 'Join Waitlist'},
+            'error': {cls: 'cta-btn--missing', cta: 'No eBook'}
+        };
+
+        var addWork = function(work) {
+            var availability = work.availability.status;
+            var ocaid = work.availability.identifier;
+            var cls = availabilityStatuses[availability].cls;
+            var url = (cls == 'cta-btn--available') ?
+                ('/borrow/ia/' + ocaid) : (cls == 'cta-btn--unavailable') ?
+                    ('/books/' + work.availability.openlibrary_edition) : work.key;
+            var cta = availabilityStatuses[availability].cta;
+            var isClickable = availability == 'error' ? 'disabled' : '';
+
+            return '<div class="book carousel__item slick-slide slick-active" ' +
+                '"aria-hidden="false" role="option">' +
+                '<div class="book-cover">' +
+                  '<a href="' + work.key + '" ' + isClickable + '>' +
+                    '<img class="bookcover" width="130" height="200" title="' + work.title + '" ' +
+                         'src="//covers.openlibrary.org/b/id/' + work.cover_id + '-M.jpg">' +
+                  '</a>' +
+                '</div>' +
+                '<div class="book-cta">' +
+                  '<a class="btn cta-btn ' + cls + '" href="' + url + '" data-ol-link-track="subjects" ' +
+                  'title="' + cta + ': ' + work.title + '" data-key="subjects" data-ocaid="' + ocaid + '">' + cta + '</a>' +
+                '</div>' +
+              '</div>';
+        }
+
+        $('.carousel-'+key).on('afterChange', function() {
```

Comment:
SUGGESTION: Add spaces around `+`. Also, give a name to *every* function, especially anonymous ones, for better in-browser debug messages when errors happen.